crates/rpc/src/jsonrpc/router.rs

```diff
@@ -92,9 +92,10 @@ impl RpcRouter {
         let output = match result {
             Ok(output) => output,
             Err(e) => {
-                tracing::warn!(method=%request.method, backtrace=?e, "RPC method panic'd");
-                // No error message so that the caller cannot learn to abuse this.
-                Err(RpcError::InternalError(anyhow::anyhow!("")))
+                tracing::error!(method=%request.method, backtrace=?e, "RPC method panic'd");
```

Comment:
For the most part, each RPC request *should* be fairly isolated - a panic in one *should not* impact the rest of them. And definitely should not impact the core operation of the node: sync'ing and verifying the data. *should* of course being the operative word here, nothing is perfect.

A panic in other parts of pathfinder might very well be cause for the `critical!` and exit strategy, but I would argue that doesn't apply to the RPC requests. It would be terrible if a bug let's the user send a fatal request which shuts down a node. This would be an extremely cheap DoS "bug".

We want to ensure that any of these logs do get sent to us to analyze; but at the same time we don't want node operators to suspend the node or cause a panic due to a log spam from a trivial bug (which only affects e.g. `starknet_getTransactionStatus`).

Maybe leaving it at `error` means we get more reports in? But we definitely cannot make it a fatal error.

Reply:
This discussion smells a lot like idealism vs pragmatism. `warn` is fine :)